streamz/core.py

```diff
@@ -457,22 +459,27 @@ def update(self, x, who=None):
 
 
 class filter(Stream):
-    def __init__(self, predicate, child):
+    def __init__(self, predicate, child, **kwargs):
         self.predicate = predicate
+        # this is one of a few stream specific kwargs
+        stream_name = kwargs.pop('stream_name', None)
 
-        Stream.__init__(self, child)
+        Stream.__init__(self, child, stream_name=stream_name)
```

Comment:
We should be able to just pass through kwargs directly to `Stream` without unpacking the `stream_name` keyword

Reply:
This will also help us to err properly when someone passes an unexpected keyword

Reply:
(which would be a good case to test for)

Reply:
I didn't understand exactly. First, I made a mistake. Here we should have:
```
    def filter(self, predicate, **kwargs):
        """ Only pass through elements that satisfy the predicate """
        return filter(predicate, self, **kwargs)
```
(the method should pass kwargs, and every other method as well. I will change this soon)

With that correction, did you mean that kwargs should be passed directly to `Stream` for every method or some methods? Methods like `map` can receive arbitrary kwargs. I don't think we can do that. Please let me know if I misunderstood. thnaks!

Reply:
I agree that with `map` or `scan` we will need to handle this differently.  In most cases though we can pass through kwargs explicitly.  We want to avoid leaving kwargs dangling.  The following should hold.

```python
def test_bad_kwarg_raises():
    source = Stream()
    with pytest.raises(TypeError):
        source.filter(lambda x: x > 1, bad_kwarg=True)
```